tests/lib/rules/destructuring-assignment.js

```diff
@@ -459,7 +459,7 @@ ruleTester.run('destructuring-assignment', rule, {
     },
     {
       code: `
-        var Hello = React.createClass({
+        var Hello = createReactClass({
           render: function() {
             return <Text>{this.props.foo}</Text>;
           }
```

Comment:
This should not be reported because `React.createClass` is **not** the correct pragma here. So there should be no component detected thus no report.